client/lib/js/react/components/ledgerplugin.js

```diff
@@ -0,0 +1,142 @@
+import EventEmitter2 from 'eventemitter2/';
+import SimpleWebsocket from 'simple-websocket/';
+
+import request from '../../utils/request';
+
+class BigchainDBLedger extends EventEmitter2 {
+
+    constructor(options) {
+        super();
+
+        this.id = null;
+        this.credentials = options.auth;
+        this.config = options.config;
+
+        this.connection = null;
+        this.connected = false;
+    }
+
+    connect() {
+        return co(this._connect.bind(this));
+    }
+
+    * _connect() {
+        const accountUri = this.credentials.account.uri;
+
+        if (this.connection) {
+            console.warn('already connected, ignoring connection request');
+            return Promise.resolve(null);
+        }
+
+        const streamUri = accountUri + '/changes';
+        console.log('subscribing to ' + streamUri);
+
+        const reconnect = reconnectCore(() => {
+            return new SimpleWebsocket(streamUri);
+        });
+
+        return new Promise((resolve, reject) => {
+            this.connection = reconnect({immediate: true}, (ws) => {
+                ws.on('open', () => {
+                    console.log('ws connected to ' + streamUri);
+                })
+                ws.on('data', (msg) => {
+                    const notification = JSON.parse(msg);
+                    co.wrap(this._handleNotification)
+                        .call(this, notification)
+                        .catch((err) => {
+                            console.warn('failure while processing notification: ' +
+                            (err && err.stack) ? err.stack : err)
+                        });
+                });
+                ws.on('close', () => {
+                    console.log('ws disconnected from ' + streamUri);
+                });
+            })
+                .once('connect', () => resolve(null))
+                .on('connect', () => {
+                    this.connected = true;
+                    this.emit('connect');
+                })
+                .on('disconnect', () => {
+                    this.connected = false;
+                    this.emit('disconnect');
+                })
+                .on('error', (err) => {
+                    console.warn('ws error on ' + streamUri + ': ' + err);
+                    reject(err);
+                })
+                .connect();
+        });
+    }
+
+    disconnect() {
+        if (this.connection) {
+            this.connection.disconnect();
+            this.connection = null;
+        }
+    }
+
+    isConnected() {
+        return this.connected;
+    }
+
+    validateTransfer(transfer) {
+        // validator.validate('TransferTemplate', transfer)
+    }
+
+    getInfo() {
+    }
+
+    getAccount() {
+        return this.credentials.account;
+    }
+
+    getBalance() {
+        return co.wrap(this._getBalance).call(this);
+    }
+
+    * _getBalance() {
+        const creds = this.credentials;
+        let res;
+        try {
```

Comment:
It might be better if we make the `request` function here purely promise-based, so that we do not need to rely on try-catch + .catch:https://github.com/bigchaindb/bigchaindb-examples/blob/master/client/lib/js/utils/request.js#L18

/cc @SohKai